crates/cli/benches/benchmark.rs

```diff
@@ -34,46 +44,42 @@ impl Function {
         let wasm_path = function_dir.join("index.wasm");
         execute_javy(&function_dir.join(js_path), &wasm_path)?;
 
-        Ok(Function {
+        let engine = Engine::default();
+        let wasm_bytes = fs::read(wasm_path)?;
+
+        let precompiled_elf_bytes = if precompiled {
+            Some(Module::new(&engine, &wasm_bytes)?.serialize()?)
+        } else {
+            None
+        };
```

Comment:
Can we add different constructors here (`Function::precompiled` and `Func::new`) instead of passing  a boolean flag? Without enough context on the underlying implementation it might be hard to tell what the flag is for.

Reply:
The reason I structured it like this is to avoid a combinatorial explosion every time we add a new dimension we want to test on.

E.g., if we were to add a third dimension called `forked` or `unforked` (just an example because I can't think of what the next dimension will be) I don't want

```rust
FunctionCase::new_static_uncompiled_nonforking(...)?;
FunctionCase::new_static_uncompiled_nonforking(...)?;
FunctionCase::new_dynamic_uncompiled_nonforking(...)?;
FunctionCase::new_dynamic_uncompiled_nonforking(...)?;
FunctionCase::new_static_precompiled_nonforking(...)?;
FunctionCase::new_static_precompiled_nonforking(...)?;
FunctionCase::new_dynamic_precompiled_nonforking(...)?;
FunctionCase::new_dynamic_precompiled_nonforking(...)?;
FunctionCase::new_static_uncompiled_forking(...)?;
FunctionCase::new_static_uncompiled_forking(...)?;
FunctionCase::new_dynamic_uncompiled_forking(...)?;
FunctionCase::new_dynamic_uncompiled_forking(...)?;
FunctionCase::new_static_precompiled_forking(...)?;
FunctionCase::new_static_precompiled_forking(...)?;
FunctionCase::new_dynamic_precompiled_forking(...)?;
FunctionCase::new_dynamic_precompiled_forking(...)?;
```

because it's super easy to miss a case and not even realize it.

What I'm open to is switching it to an enum instead of a bool since that's a totally fair criticism that it's difficult to determine what the boolean represents!

So we could do something like:

```rust
for linking in [Linking::Static, Linking::Dynamic] {
  for compilation in [Compilation::Precompiled, Compilation::Uncompiled] {
    for forking in [Forking::Forking, Forking::NonForking] {
      FunctionCase::new(..., linking, compilation, forking)?;
      FunctionCase::new(..., linking, compilation, forking)?;
    }
  }
}
```

Reply:
Oh makes sense, I misread that!